static/js/unread.js

```diff
@@ -413,6 +459,32 @@ exports.set_read_flag = function (message) {
     message.unread = false;
 };
 
+exports.load_server_counts = function () {
+    var unread_msgs = page_params.unread_msgs;
+
+    exports.unread_pm_counter.set_huddles(unread_msgs.huddles);
+    exports.unread_pm_counter.set_pms(unread_msgs.pms);
+    exports.unread_topic_counter.set_streams(unread_msgs.streams);
+    exports.unread_mentions_counter.add_many(unread_msgs.mentions);
+
+    _.each(unread_msgs.huddles, function (obj) {
+        unread_messages.add_many(obj.unread_message_ids);
+    });
+    _.each(unread_msgs.pms, function (obj) {
+        unread_messages.add_many(obj.unread_message_ids);
+    });
+    _.each(unread_msgs.streams, function (obj) {
+        unread_messages.add_many(obj.unread_message_ids);
+    });
+    unread_messages.add_many(unread_msgs.mentions);
+};
+
+exports.initialize = function () {
+    if (feature_flags.load_server_counts) {
+        exports.load_server_counts();
+    }
```

Comment:
Because the change in this case is something that happens at initialization time, I'm not sure there's actually a way to trigger this to happen at the same time it'd happen on page load.

I wonder if we want to have a mechanism to set `feature_flags` based on some extra parameters added to the URL or something.